blog/2019-07-04-tutorial-sentry-debezium-docker.adoc

```diff
@@ -0,0 +1,135 @@
+
+log4j.appender.appender=org.apache.log4j.DailyRollingFileAppender
+log4j.appender.appender.DatePattern='.'yyyy-MM-dd-HH
+log4j.appender.appender.File=${kafka.logs.dir}/connect-service.log
+log4j.appender.appender.layout=org.apache.log4j.PatternLayout
+log4j.appender.appender.layout.ConversionPattern=%d{ISO8601} %-5p  %X{dbz.connectorType}|%X{dbz.connectorName}|%X{dbz.connectorContext}  %m   [%c]%n
+
+log4j.appender.Sentry=io.sentry.log4j.SentryAppender
+log4j.appender.Sentry.threshold=WARN
+
+----
+
+=== Sentry.io SSL certificate
+
+https://docs.sentry.io/ssl/ download the `getsentry.pem` file and put it in your project's directory under `ssl/` directory.
+
+=== The Dockerfile
+
+Now we can glue everything together in our Debezium image:
+- Let's first create a JKS file with our sentry certificate, this uses a Docker multi-stage building process, where we are generating a `certificates.jks` which later we copy into our Kafka Connect with Debezium stage
+- Copy `log4j.properties` into `$KAFKA_HOME/config/log4j.properties`
+- Copy the jks from the multi-stage build
+- Set ENV with the Sentry version and m5sums
+- Download Sentry dependencies, the script you see called `docker-maven-download` is a helper which we ship by default in our images, in this case we're setting it to download a jar file from the maven central and put it in the Kafka libs directory, we do that by setting the ENV var `MAVEN_DEP_DESTINATION=$KAFKA_HOME/libs`
+
+[source,dockerfile,docker]
+----
+FROM fabric8/java-centos-openjdk8-jdk:1.6 as ssl-jks
+
+ARG JKS_STOREPASS="any random password, you can also set it outside via the arguments from docker build"
+
+USER root:root
+
+COPY /ssl /ssl
+
+RUN chown -R jboss:jboss /ssl
+
+USER jboss:jboss
+
+WORKDIR /ssl  
+
+RUN keytool -import -noprompt -alias getsentry \
+    -storepass "${JKS_STOREPASS}" \
+    -keystore certificates.jks \
+    -trustcacerts -file "/ssl/getsentry.pem"
+
+FROM debezium/connect:0.10 AS kafka-connect
+
+EXPOSE 8083
+
+COPY config/log4j.properties "$KAFKA_HOME/config/log4j.properties"
+
+COPY --from=ssl-jks --chown=kafka:kafka /ssl/certificates.jks /ssl/
+
+ENV SENTRY_VERSION=1.7.23 \
+    MAVEN_DEP_DESTINATION=$KAFKA_HOME/libs
+
+RUN docker-maven-download central io/sentry sentry "$SENTRY_VERSION" 4bf1d6538c9c0ebc22526e2094b9bbde && \
+    docker-maven-download central io/sentry sentry-log4j "$SENTRY_VERSION" 74af872827bd7e1470fd966449637a77
+
+----
+
+
+=== Build and run
+
+Now we can simply build the image:
+
+
+[source,bash]
+----
+$ docker build -t debezium/connect-sentry:1 --build-arg=JKS_STOREPASS="123456789" .
+----
+
+
+When running the image we have now to configure our Kafka Connect application to load the JKS by setting `KAFKA_OPTS: -Djavax.net.ssl.trustStore=/ssl/certificates.jks -Djavax.net.ssl.trustStorePassword=123456789`
+
+In case you don't know where to put this variable or how to setup our Docker setup, please https://debezium.io/docs/tutorial/#starting_docker[check our tutorial].
```

Comment:
```suggestion
In case you'd like to learn more about using the Debezium container images, please link:/docs/tutorial/#starting_docker[check our tutorial].
```